core/models/request_matcher.go

```diff
@@ -313,10 +313,15 @@ func (this RequestMatcher) ToEagerlyCacheable() *RequestDetails {
 	if this.Query != nil && len(*this.Query) > 0 {
 		for key, valueMatchers := range *this.Query {
 			for _, valueMatcher := range valueMatchers {
-				if valueMatcher.Matcher != matchers.Exact {
+				if valueMatcher.Matcher != matchers.Exact && !(valueMatcher.Matcher == matchers.Array && (valueMatcher.Config == nil || len(valueMatcher.Config) == 0)) {
 					return nil
 				}
-				query[key] = []string{valueMatcher.Value.(string)}
+				if valueMatcher.Matcher == matchers.Array {
+					query[key] = valueMatcher.Value.([]string)
```

Comment:
this line throws an array when I import a simulation with query array matcher: 
```
panic: interface conversion: interface {} is []interface {}, not []string

goroutine 1 [running]:
github.com/SpectoLabs/hoverfly/core/models.RequestMatcher.ToEagerlyCacheable({{0x140003e4810, 0x1, 0x1}, {0x140003e4840, 0x1, 0x1}, {0x140003e4870, 0x1, 0x1}, {0x140003e48a0, ...}, ...})
        /Users/tommy.situ/repos/kapish/hoverfly/core/models/request_matcher.go:320 +0x958
github.com/SpectoLabs/hoverfly/core/matching.(*CacheMatcher).PreloadCache(0x140002efd60, 0x16dd5b5f2?)
        /Users/tommy.situ/repos/kapish/hoverfly/core/matching/cache_matcher.go:148 +0xdc
main.main()
        /Users/tommy.situ/repos/kapish/hoverfly/core/cmd/hoverfly/main.go:554 +0x2ee4

```

You can try capture a simulation using this:
`curl --proxy localhost:8500 'time.jsontest.com?token=1&token=2'`

Reply:
incorporated